Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -8841,6 +8841,73 @@ def shorten_string_for_printing(source_string, max_length=64):
         return ret_value
 
 
+class PollResult:
+    def __init__(self, response, continue_to_poll=False, args_for_next_run=None):
+        """
+        Constructor for PollResult
+
+        :type response: ``Any``
+        :param response:  The response of the command in the event of success,
+        or in case of failure but Polling is false
+
+        :type continue_to_poll: ``Union[bool, Callable]``
+        :param continue_to_poll: an iterable of relevant keys list from the json. Notice we save it as a set in the class
+
+        :type args_for_next_run: ``Dict``
+        :param args_for_next_run: The arguments to use in the next iteration. Will use the input args in case of None
+
+        """
+        self.response = response
+        self.continue_to_poll = continue_to_poll
+        self.args_for_next_run = args_for_next_run
+
+
+def poll(name, interval=30, timeout=600, poll_message='Fetching Results:', polling_arg_name="Polling"):
+    """
+    To use on a function that should rerun itself
+    Commands that use this decorator must have a Polling argument, polling: true in yaml,
+    and a hidden polled_once argument.
```

Comment:
Let's make sure to explain that this property effectively hides the `readalbe_output`. it might be better to call it `hide_polling_output` or something along that line.